blockchain/types/src/block.rs

```diff
@@ -192,6 +207,11 @@ pub fn compute_block_id(
     root_element.append_to_transcript(b"root_element", &mut transcript);
     contents_hash.append_to_transcript(b"contents_hash", &mut transcript);
 
+    // Earlier blocks didn't have a timestamp so it will come back as 0
+    if timestamp != 0 {
```

Comment:
Might be better to rely on the block version for this

Reply:
Fixed in #3733